std/datetime.d

```diff
@@ -1075,6 +1075,25 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        if (_stdTime < 0)
+            return (cast(size_t) (_stdTime * -1)) + long.max;
```

Comment:
> Actually, this is still going to fail on 32-bit systems (result of size_t + long is long on 32-bit). I think the most straightforward thing to do is just cast(size_t).

This will be very bad hash for 32-bit machines. The better one is use mixing from MurmurHash2 (not 3) for 64 bit integers, and then cast it to `size_t`